src/XR/features/WebXRHandTracking.ts

```diff
@@ -324,112 +367,196 @@ export class WebXRHand implements IDisposable {
      */
-    public get handMesh() {
-        return this._handMesh;
+    public setHandMesh(handMesh: AbstractMesh, rigMapping: Nullable<XRHandMeshRigMapping>) {
+        this._handMesh = handMesh;
+
+        // Link the bones in the hand mesh to the transform nodes that will be bound to the WebXR tracked joints.
+        if (this._handMesh.skeleton) {
+            const handMeshSkeleton = this._handMesh.skeleton;
+            handJointReferenceArray.forEach((jointName, jointIdx) => {
+                const jointBoneIdx = handMeshSkeleton.getBoneIndexByName(rigMapping ? rigMapping[jointName] : jointName);
+                if (jointBoneIdx !== -1) {
+                    handMeshSkeleton.bones[jointBoneIdx].linkTransformNode(this._jointTransforms[jointIdx]);
+                }
+            });
+        }
     }
 
     /**
-     * Update this hand from the latest xr frame
-     * @param xrFrame xrFrame to update from
-     * @param referenceSpace The current viewer reference space
-     * @param scaleFactor optional scale factor for the meshes
+     * Update this hand from the latest xr frame.
+     * @param xrFrame The latest frame received from WebXR.
+     * @param referenceSpace The current viewer reference space.
      */
-    public updateFromXRFrame(xrFrame: XRFrame, referenceSpace: XRReferenceSpace, scaleFactor: number = 2) {
+    public updateFromXRFrame(xrFrame: XRFrame, referenceSpace: XRReferenceSpace) {
         const hand = this.xrController.inputSource.hand;
         if (!hand) {
             return;
         }
-        this.trackedMeshes.forEach((mesh, name) => {
-            const xrJoint = hand.get(name);
-            if (xrJoint) {
-                let pose = xrFrame.getJointPose!(xrJoint, referenceSpace);
-                if (!pose || !pose.transform) {
-                    return;
-                }
-                // get the transformation. can be done with matrix decomposition as well
-                const pos = pose.transform.position;
-                const orientation = pose.transform.orientation;
-                mesh.position.set(pos.x, pos.y, pos.z);
-                mesh.rotationQuaternion!.set(orientation.x, orientation.y, orientation.z, orientation.w);
-                // left handed system conversion
-                // get the radius of the joint. In general it is static, but just in case it does change we update it on each frame.
-                const radius = (pose.radius || 0.008) * scaleFactor;
-                mesh.scaling.set(radius, radius, radius);
-
-                // if we are using left-handed meshes, transform before applying to the hand mesh
-                if (this._leftHandedMeshes && !mesh.getScene().useRightHandedSystem) {
-                    mesh.position.z *= -1;
-                    mesh.rotationQuaternion!.z *= -1;
-                    mesh.rotationQuaternion!.w *= -1;
+
+        const anyHand: any = hand;

```

Comment:
just wondering - why is any needed? is the type not enough?

Reply:
Seems that TypeScript doesn't allow you to override the indexing ([]) operator, so I need to convert to any to allow dynamic access of members on the object

Reply:
I've spent a lot of time looking into this comment, and I think we're going to have to take a more in-depth look at the way we're interpreting the [WebXR spec IDL](https://immersive-web.github.io/webxr-hand-input/#xrhand-interface) for XRHand in webxr.d.ts, especially with the new changes to the spec​. 

I'm going to file an issue about exactly what I mean, but for now I think I'm going to have to leave the ​any​ cast as-is because otherwise I would have to make a bunch of crazy changes to our .d.ts.

Reply:
I am prefectly fine with adjusting our webxr.d.ts to fit the IDL. But you can do that in a future PR. if you do, please put a TODO above this line so we know it's not the final state